src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,76 @@
-  return (
-    <div className="modal is-active" data-cy="modal">
-      <div className="modal-background" />
+interface TodoModalProps {
+  onClose: () => void;
+  user: User | null;
+}
 
-      <Loader />
-
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
+export const TodoModal: React.FC<TodoModalProps> = ({ onClose, user }) => {
+  const currentTodo = useSelector((state: RootState) => state.currentTodo);
+  const [loading, setLoading] = useState(true);
 
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button type="button" className="delete" data-cy="modal-close" />
-        </header>
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      if (currentTodo) {
+        setLoading(false);
+      }
+    }, 300);
 
-        <div className="modal-card-body">
-          <p className="block" data-cy="modal-title">
-            fugiat veniam minus
-          </p>
+    return () => clearTimeout(timer);
+  }, [currentTodo]);
 
-          <p className="block" data-cy="modal-user">
-            {/* For not completed */}
-            <strong className="has-text-danger">Planned</strong>
+  if (!currentTodo) {
+    return null;
+  }
 
-            {/* For completed */}
-            <strong className="has-text-success">Done</strong>
-            {' by '}
-            <a href="mailto:Sincere@april.biz">Leanne Graham</a>
-          </p>
+  return (
+    <div className="modal is-active" data-cy="modal">
+      <div className="modal-background" onClick={onClose} />
+      {loading ? (
+        <div className="loader-container">
+          <Loader />
+        </div>
+      ) : (
+        <div className="modal-card">
+          <header className="modal-card-head">
+            <div
+              className="modal-card-title has-text-weight-medium"
+              data-cy="modal-header"
+            >
+              Todo #{currentTodo.id}
+            </div>
+            <button
+              type="button"
+              className="delete"
+              data-cy="modal-close"
+              onClick={onClose}
+            />
+          </header>
+          <div className="modal-card-body">
+            <p className="block" data-cy="modal-title">
+              {currentTodo.title}
+            </p>
+            {user && (
+              <p className="block" data-cy="modal-user">
+                <strong>{user.name}</strong> {' by '}
+                <strong
+                  className={
+                    currentTodo.completed
+                      ? 'has-text-success'
+                      : 'has-text-danger'
+                  }
+                >
+                  {currentTodo.completed ? 'Done' : 'Planned'}
+                </strong>
+              </p>
+            )}
```

Comment:
This is a minor suggestion, but consider moving the user information into a separate component. This would make the code more modular and easier to maintain.